packages/components/src/components/input/input.ts

```diff
@@ -388,61 +403,129 @@ export default class SdInput extends SolidElement implements SolidFormControl {
   }
 
   render() {
-    const hasLabelSlot = this.hasSlotController.test('label');
-    const hasHelpTextSlot = this.hasSlotController.test('help-text');
-    const hasLabel = this.label ? true : !!hasLabelSlot;
-    const hasHelpText = this.helpText ? true : !!hasHelpTextSlot;
-    const hasClearIcon =
-      this.clearable && !this.disabled && !this.readonly && (typeof this.value === 'number' || this.value.length > 0);
-
+    // Slots
+    const slots = {
+      label: this.hasSlotController.test('label'),
+      helpText: this.hasSlotController.test('help-text'),
+      description: this.hasSlotController.test('description'),
+      left: this.hasSlotController.test('left'),
+      right: this.hasSlotController.test('right')
+    };
+
+    // States
+    const hasLabel = this.label ? true : !!slots['label'];
+    const hasHelpText = this.helpText ? true : !!slots['helpText'];
+    const hasClearIcon = this.clearable && !this.readonly && (typeof this.value === 'number' || this.value.length > 0);
+    const hasValidationAttr = this.required || !!this.pattern || !!this.minlength || !!this.maxlength;
+    const isInvalid = hasValidationAttr && !this.checkValidity();
+    const isValid = hasValidationAttr && this.checkValidity();
```

Comment:
I have just noticed that the basically same constant appears twice here. Moreover, if `hasValidationAttr = false`, then `isInvalid = false` and `isValid = false` too, which makes no sense.

Reply:
isInvalid is used to hide or show the "risk" icon
isValid is used to hide or show the "success" icon
![image](https://github.com/solid-design-system/solid/assets/39494579/ac07de06-ee52-4e2b-b9d5-21e57e401f31)

IF there is no validation attribute, then there shouldn't be either type of validation icon present.  This makes sense to me.  How would you change this to make it more readable / understandable? I'm open to suggestions.  However, the 'success' state does not really make sense to me.  I included a TODO comment which was intended to stimulate conversation:

line 630:
<!-- TODO: How are the "success" styles supposed to work? Is this sufficient? -->

I'm a little frustrated that my comments are still being ignored and I am only offered cherry-picked feedback here.  I will bring this up in the daily tomorrow.

Reply:
@saemik94

Reply:
Ahh, now I understand what's going on, it's more difficult than I thought. Thanks for the explanation